apps/vs-code-designer/src/app/utils/funcCoreTools/getFuncPackageManagers.ts

```diff
@@ -2,36 +2,73 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
-import { funcPackageName, PackageManager } from '../../../constants';
-import { tryGetInstalledBrewPackageName } from '../packageManagers/getBrewPackageName';
+import { funcPackageName, PackageManager, Platform, wingetFuncPackageName } from '../../../constants';
+import {
+  getFuncCoreToolsBrewPackageName,
+  isBrewPackageInstalled,
+  tryGetInstalledBrewPackageName,
+} from '../packageManagers/getBrewPackageName';
 import { executeCommand } from './cpUtils';
 import { FuncVersion } from '@microsoft/vscode-extension';
 
 /**
  * Gets package managers installed in the system.
+ * Determines which package manager installed functions core tools.
  * @param {boolean} isFuncInstalled - Is functions core tools installed.
  * @returns {Promise<PackageManager[]>} Returns array of package managers.
  */
 export async function getFuncPackageManagers(isFuncInstalled: boolean): Promise<PackageManager[]> {
   const result: PackageManager[] = [];
-  switch (process.platform) {
-    case 'linux':
-      // https://github.com/Microsoft/vscode-azurefunctions/issues/311
-      break;
-    case 'darwin':
-      if (await hasBrew(isFuncInstalled)) {
+  if (isFuncInstalled) {
+    if (process.platform == Platform.windows) {
+      const response = await executeCommand(undefined, undefined, 'winget', 'list', wingetFuncPackageName);
+      // winget response includes headers if package is installed
+      if (response.length > 2) {
+        result.push(PackageManager.winget);
+      }
+    }
+
+    if (process.platform == Platform.mac) {
+      if (
+        isBrewPackageInstalled(getFuncCoreToolsBrewPackageName(FuncVersion.v4)) ||
```

Comment:
Does the hasBrew(isFuncInstalled) doesn't do this?

Reply:
true